src/BaseCollection.php

```diff
@@ -112,14 +112,14 @@ public function filter(callable $callback = null, int $flag = 0): BaseCollection
 	 */
 	public function reverse(): BaseCollection
 	{
-		return new static(array_reverse($this->getArrayCopy()), $this->getTotalCount());
+		return new self(array_reverse($this->getArrayCopy()), $this->getTotalCount());
```

Comment:
This is wrong, as it will return a `BaseCollection` instance when an inherited class instance is expected instead. If the return type hint allows for `static`, then it should be used instead.

Reply:
fixed.